core/lib/basic_types/src/tee_types.rs

```diff
@@ -1,9 +1,58 @@
+use std::{fmt, str::FromStr};
+
 use serde::{Deserialize, Serialize};
-use strum::{Display, EnumString};
 
-#[derive(Debug, Clone, Copy, PartialEq, EnumString, Display, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
 #[non_exhaustive]
 pub enum TeeType {
-    #[strum(serialize = "sgx")]
     Sgx,
 }
+
+impl fmt::Display for TeeType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TeeType::Sgx => write!(f, "sgx"),
+        }
+    }
+}
+
+impl FromStr for TeeType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_json::from_str(&format!("\"{}\"", s.to_lowercase()))
+            .map_err(|_| format!("Invalid value for TeeType: {}", s))
+    }
```

Comment:
Sorry for many iterations of review, but:
1. Why do we need to allocate a string and then deserialize it just to get the value? Why not check on static strings?
2. It is somewhat unconventional to have case-insensitive logic in `FromStr`. I would prefer the logic to be strict, and if we don't know if the string is lowercased, then the _caller_ of this method can call `TeeType::from_str(s.to_lowercase())`.

Reply:
I completely removed the implementation of the `std::str::FromStr` trait because we don't use it anyway.